### DISCUSSION TITLE
Can I remove the 'click' functions assigned to buttons/icons?

### DISCUSSION THREAD

**Discussion by Anindya088** [zauberzeug/nicegui#1699] [Q&A] [answered]
### Question

Hi All,

I have an icon where I assign a 'click' function using the ```icon.on``` method. This is just a simple click to go to a page. 
Now let's say I want to remove that action, how do I remove that click function go away? Meaning even if I click the icon now, nothing should happen. Is there a way to do this?

This is what I have tried so far, in a sketch:
```python
# Sample page for testing
@ui.page('/pageA')
def functionA():
    ui.label('Test page A').classes('text-[100px] text-red')

# This is the test icon where I assign the page
icon1 = ui.icon('person',size='xl').classes('cursor-pointer')
icon1.on('click', lambda: ui.open('/pageA'))

# I want to use this icon to remove the 'click' event from first icon
icon2 = ui.icon('settings',size='xl').classes('cursor-pointer')
icon2.on('click', lambda: update_icon1())

# Function to remove 'click' event from icon1
async def update_icon1():
    app.remove_route('/pageA')
    icon1.classes(remove='cursor-pointer', add='cursor-default')

    js_code = f'''
        var element = getElement({icon1.id});
        element.addEventListener("click", function(event) {{
            // return false ;
            event.preventDefault();
        }});
        '''
    await ui.run_javascript(js_code, respond=False)
```

I was thinking, since the 'click' functions are js functions, so I will try some js code to remove them. But I think I am wrong, either in the approach or in the actual js code. 

Now, what's happening is I can still click on icon1, and it still goes to that 'pageA' -- just that its removed so it gives an error.
But I want to make it such that even if I click it, nothing happens anymore.

What should be the proper way to do this? Kindly advise me on this.

Thanks a lot,
Anindya

**Comment by natankeddem** [comment, marked as answer]
I can't see an exposed way to remove an event once registered. I guess you could manipulate `_event_listeners` if you knew which event you wanted to remove. Maybe I am missing something as well, I didn't search that long. See related discussion https://github.com/zauberzeug/nicegui/discussions/1516. Do you actually need to stop the event or do you just want to stop executing code in reaction to the clicking? Also, you might want to look into using `ui.button` it has `click` event registration cooked in.

```py
from nicegui import ui

click_active = True

def clicked():
    global click_active
    if click_active:
        ui.notify("click")

def toggle_icon_click():
    global click_active
    click_active = not click_active
    ui.notify(f"click_active = {click_active}")


ui.button(icon="person", on_click=clicked)
ui.button("Toggle Icon Click", on_click=toggle_icon_click)
ui.run()
```
You can also lambda the evaluation instead:
```py
ui.button(icon="person", on_click=lambda _:  clicked() if click_active else None)
```

**Reply by falkoschindler** [reply]
I agree. Feature request https://github.com/zauberzeug/nicegui/discussions/1516 is indeed closely related.
Currently a flag like `click_active` is the way to go.

**Reply by Anindya088** [reply]
Hi @natankeddem , @falkoschindler ,

Thanks so much for the detailed answer, very clear now. Sorry I was offline for few days, could not respond earlier.
I will go with the global flag as you advised, thanks 😄 

Regards,
Anindya
